app/controllers/submissions_controller.rb

```diff
@@ -346,7 +346,8 @@ def update_files
         return
       end
     else
-      @grouping = @assignment.groupings.find(params[:grouping_id])
+      @grouping = @assignment.groupings.find(params[:grouping_id]) ||
+          @assignment.groupings.joins(student_memberships: :user).where('users.type': 'TestStudent').first
```

Comment:
This should not be necessary, the `params[:grouping_id]` value should still be correct